meagea-api/src/main/java/project/controller/PromotionController.java

```diff
@@ -19,29 +24,89 @@
 public class PromotionController {
 
     private final PromotionService proService;
+    private final AnimalService animalService;
+    private final LogService logService;
 
     @PostMapping("/promotion")
-    public PromotionDetailDto addPromotion(@ModelAttribute PromotionForm form) throws IOException {
-        return proService.savePromotion(form);
+    public PromotionDetailDto addPromotion(@ModelAttribute PromotionForm form) throws Exception {
+        Promotion pro = proService.savePromotion(form);
+        Animal animal = animalService.findAnimalByNo(form.getAnimalNo());
+
+        List<AnimalFile> animalFileList = new ArrayList<>();
+        CompletableFuture<AnimalFile> fileFuture = proService.saveAnimalFile(pro.getNo(), form.getImageList());
+        CompletableFuture<List<AnimalFile>> fileListFuture = fileFuture.thenApplyAsync(file -> proService.turnAnimalList(file, animalFileList));
```

Comment:
```suggestion
        CompletableFuture<List<AnimalFile>> fileListFuture = fileFutures.thenApplyAsync(files -> proService.turnAnimalList(files, animalFileList));
```

위 처럼하거나, 아니면, async안에서 for-loop을 돌거나 할 수 있겠죠.